compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/ProjectLoader.java

```diff
@@ -95,31 +136,41 @@ public static Project loadProject(Path path, ProjectEnvironmentBuilder projectEn
      * @return documentId of the document
      */
 
-    public static Optional<DocumentId> getDocumentId(Path documentFilePath, BuildProject project) {
+    public static Document getDocument(Path documentFilePath, Project project) {
```

Comment:
IMO adding an overloaded API so that the users can retrieve the document instance just by passing its path might makes more sense from the users point of view. WDYT?

```
public static Document getDocument(Path documentFilePath) {
    Project project = loadProject(documentFilePath);
    return getDocument(documentFilePath, project);
}
```

Reply:
I think the use case tried to solve here is to get a document out of an **existing** project instance. Hence, I suggest this should be moved to `ProjectUtils`. 

Or I suppose this should be a basic-functionality of an **Project**. 

```java
public interface Project {
    Optional<Document> getDocument(Path documentFilePath);
}
```

Reply:
My gut feeling is that this method shouldn't be in the project API. Project API offers methods to retrieve modules by their names, and ids. Working with paths should not be part of the project API code.  

Can we keep a map of paths to document IDs?